Library/Homebrew/resource.rb

```diff
@@ -145,7 +145,10 @@ def url(val = nil, specs = {})
   end
 
   def version(val = nil)
-    @version ||= detect_version(val)
+    @version ||= begin
+      version = detect_version(val)
+      version.is_a?(Version::NullVersion) ? nil : version
```

Comment:
Maybe this is too strict but I see explicitly checking for `is_a?` to be a bit of a smell; particularly when converting to `nil` so would be 🆒 to try and avoid them when possible.